src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/ObjectWriter/R2RPEBuilder.cs

```diff
@@ -596,11 +600,13 @@ public static PEHeaderBuilder Copy(PEHeaders peHeaders, TargetDetails target)
                 sectionAlignment = fileAlignment;
             }
 
-            DllCharacteristics dllCharacteristics = DllCharacteristics.DynamicBase | DllCharacteristics.NxCompatible;
+            dllCharacteristics &= (DllCharacteristics.TerminalServerAware | DllCharacteristics.AppContainer);
+            dllCharacteristics |= DllCharacteristics.DynamicBase | DllCharacteristics.NxCompatible;
 
-            // Copy over selected DLL characteristics bits from IL image
-            dllCharacteristics |= peHeaders.PEHeader.DllCharacteristics &
-                (DllCharacteristics.TerminalServerAware | DllCharacteristics.AppContainer);
```

Comment:
In general, the changes in here look meaningful, and I'm not sure why you made them. Please work with Simon to make sure these are the right changes to make.

Reply:
I use https://github.com/dotnet/runtime/blob/b8591b4a9ce88f98ff6aba93241f23e1512c01fa/src/coreclr/src/zap/zapimage.cpp#L1511 as the source of truth for these values so we behave the same as Crossgen1

Reply:
I have compared the Crossgen2 code to the bit in zapimage.cpp you pointed out and I fixed a few small differences but I believe that even before we were mostly doing the same thing.